internal/controllers/topology/cluster/conditions.go

```diff
@@ -138,130 +146,172 @@ func (r *Reconciler) reconcileTopologyReconciledCondition(s *scope.Scope, cluste
 		return nil
 	}
 
-	// If any of the lifecycle hooks are blocking any part of the reconciliation then topology
-	// is not considered as fully reconciled.
-	if s.HookResponseTracker.AggregateRetryAfter() != 0 {
+	// If the cluster is performing initial provisioning.
+	if s.Blueprint != nil && s.Blueprint.ClusterClass != nil &&
+		s.Blueprint.ClusterClass.GetGeneration() != s.Blueprint.ClusterClass.Status.ObservedGeneration {
 		v1beta1conditions.Set(cluster,
 			v1beta1conditions.FalseCondition(
 				clusterv1.TopologyReconciledV1Beta1Condition,
-				clusterv1.TopologyReconciledHookBlockingV1Beta1Reason,
+				clusterv1.TopologyReconciledClusterClassNotReconciledV1Beta1Reason,
 				clusterv1.ConditionSeverityInfo,
-				// TODO: Add a protection for messages continuously changing leading to Cluster object changes/reconcile.
-				"%s", s.HookResponseTracker.AggregateMessage(),
+				"ClusterClass not reconciled. If this condition persists please check ClusterClass status. A ClusterClass is reconciled if"+
+					".status.observedGeneration == .metadata.generation is true. If this is not the case either ClusterClass reconciliation failed or the ClusterClass is paused",
 			),
 		)
 		conditions.Set(cluster, metav1.Condition{
 			Type:   clusterv1.ClusterTopologyReconciledCondition,
 			Status: metav1.ConditionFalse,
-			Reason: clusterv1.ClusterTopologyReconciledHookBlockingReason,
-			// TODO: Add a protection for messages continuously changing leading to Cluster object changes/reconcile.
-			Message: s.HookResponseTracker.AggregateMessage(),
+			Reason: clusterv1.ClusterTopologyReconciledClusterClassNotReconciledReason,
+			Message: "ClusterClass not reconciled. If this condition persists please check ClusterClass status. A ClusterClass is reconciled if" +
+				".status.observedGeneration == .metadata.generation is true. If this is not the case either ClusterClass reconciliation failed or the ClusterClass is paused",
 		})
 		return nil
 	}
 
-	// The topology is not considered as fully reconciled if one of the following is true:
-	// * either the Control Plane or any of the MachineDeployments/MachinePools are still pending to pick up the new version
-	//  (generally happens when upgrading the cluster)
-	// * when there are MachineDeployments/MachinePools for which the upgrade has been deferred
-	// * when new MachineDeployments/MachinePools are pending to be created
-	//  (generally happens when upgrading the cluster)
+	// If the BeforeClusterCreate hook is blocking, reports it
+	if !s.Current.Cluster.Spec.InfrastructureRef.IsDefined() && !s.Current.Cluster.Spec.ControlPlaneRef.IsDefined() {
+		if s.HookResponseTracker.AggregateRetryAfter() != 0 {
```

Comment:
Is it really intentional that we say topology is reconciled when we're stuck in BeforeClusterCreate?

(I'm not sure why the condition should be true in that case, while it is false when the upgrades or the deletion is blocked)

Reply:
I aligned this to the same consideration we did for upgrade hooks. See also PR description 

> hooks are integral part of the cluster upgrade workflow (or of the cluster provisioning/deletion workflows).